lib/core/src/examples/easing_animator.rs

```diff
@@ -0,0 +1,331 @@
+        let point = properties.position;
+        self.context.save();
+        self.context.set_fill_style(&color.into());
+        self.context.scale(self.width() / 2.0, self.height() / 2.0).ok();
+        self.context.set_line_width(2.0 / self.height());
+        self.context.translate(1.0, 1.0).ok();
+        self.context.fill_rect(point.x as f64 - size / 2.0,point.y as f64 - size / 2.0,size,size);
+        self.context.restore();
+    }
+
+    /// Draw a 2D graph of the provided FnEasing function.
+    pub fn graph<F:FnEasing>(&self, f:F, color:&str, time_ms:f64) {
+        let width  = self.width() - 1.0;
+        let height = self.height();
+
+        self.context.set_stroke_style(&color.into());
+        self.context.begin_path();
+        self.context.save();
+        self.context.scale(width, height / 2.0).ok();
+        self.context.translate(0.0, 0.5).ok();
+        self.context.set_line_width(1.0 / height);
+        self.context.move_to(0.0, f(0.0));
+        for x in 1..self.canvas.width() {
+            let x = x as f64 / width;
+            let y = f(x);
+            self.context.line_to(x, y);
+        }
+        self.context.stroke();
+
+        self.context.set_fill_style(&color.into());
+        let width  = 8.0  / width;
+        let height = 16.0 / height;
+        let time_seconds = time_ms / 1000.0;
+        let x      = (time_seconds % 3.0) / 2.0;
+        let y      = f(x);
+        self.context.fill_rect(x - width / 2.0, y - height / 2.0, width, height);
+        self.context.restore();
+    }
+}
+
+/// Creates a Vector3<f32> with random components from -1 to 1.
+fn vector2_random() -> Vector2<f32> {
+    let x = ((Math::random() - 0.5) * 2.0) as f32;
+    let y = ((Math::random() - 0.5) * 2.0) as f32;
+    Vector2::new(x, y)
+}
+
+struct SharedData {
+    graph_canvas     : Canvas,
+    animation_canvas : Canvas,
+    easing_animator  : Option<EasingAnimator<Properties>>,
+    properties       : Properties,
+    easing_function  : &'static dyn FnEasing
+}
+
+#[derive(Clone)]
+struct SubExample {
+    data : Rc<RefCell<SharedData>>
+}
+
+impl SubExample {
+    fn new<F>
+    (mut event_loop   : &mut EventLoop
+     , graph_canvas     : Canvas
+     , animation_canvas : Canvas
+     , f                : &'static F
+     , initial_value    : Properties
+     , final_value      : Properties) -> Self
+    where F:FnEasing {
+        let properties       = Properties::new(Vector2::new(0.0, 0.0), 1.0);
+        let easing_animator  = None;
+        let easing_function = f;
+        let data = SharedData {
+            properties,
+            easing_function,
+            graph_canvas,
+            animation_canvas,
+            easing_animator
+        };
+        let data = Rc::new(RefCell::new(data));
+        let weak = Rc::downgrade(&data);
+        let easing_animator = EasingAnimator::new(
+            &mut event_loop,
+            move |value| { weak.upgrade().map(|data| data.borrow_mut().properties = value); },
```

Comment:
This are examples so I'm ok with it, but in normal code, I would really dislike naked refcells like here. If you see a simple way to just wrap those in a newtype, please do. If not, lets leave them.